tools/release_do_nothing.py

```diff
@@ -870,29 +1138,56 @@ def merge_back(self):
             "preserve the commit SHA's."
         )
 
-        if self.first_in_series:
-            # TODO: automate
+        def next_series_patch() -> IrisVersion:
+            tagged_versions = self._get_tagged_versions()
+            series_all = sorted(set(v.series for v in tagged_versions))
+            try:
+                next_series = series_all[series_all.index(self.version.series) + 1]
+            except (IndexError, ValueError):
+                message = f"Error finding next series after {self.version.series} ."
+                raise RuntimeError(message)
 
-            working_branch = self.strings.branch + ".mergeback"
-            self._delete_local_branch(working_branch)
-            message = (
-                "Checkout a local branch from the official ``main`` branch.\n"
-                "git fetch upstream;\n"
-                f"git checkout upstream/main -b {working_branch};"
+            series_latest = max(
+                v for v in tagged_versions if v.series == next_series
+            )
+            return IrisVersion(
+                f"{series_latest.major}.{series_latest.minor}.{series_latest.micro + 1}"
             )
-            self.wait_for_done(message)
 
+        if self.more_patches_after_this_one:
             message = (
-                f"Merge in the commits from {self.strings.branch}.\n"
-                f"{merge_commit}\n"
-                f"git merge upstream/{self.strings.branch} --no-ff "
-                '-m "Merging release branch into main";'
+                "More series need patching. Merge into the next series' branch ..."
             )
-            self.wait_for_done(message)
+            self.print(message)
+            next_patch = next_series_patch()
+            target_branch = next_patch.branch
+            working_branch = f"{self.version}-to-{target_branch}"
+        else:
+            next_patch = None
+            target_branch = "main"
+            working_branch = self.version.branch + ".mergeback"
 
+        # TODO: automate
+        self._delete_local_branch(working_branch)
+        message = (
+            "Checkout a local branch from the official branch.\n"
+            f"git fetch {self.github_scitools};\n"
+            f"git checkout {self.github_scitools}/{target_branch} -b {working_branch};"
+        )
+        self.wait_for_done(message)
+
+        message = (
+            f"Merge in the commits from {self.version.branch}.\n"
+            f"{merge_commit}\n"
+            f"git merge {self.github_scitools}/{self.version.branch} --no-ff "
+            f'-m "Merging {self.version.branch} into {target_branch}";'
+        )
+        self.wait_for_done(message)
+
+        if self.first_in_series:
```

Comment:
This logic is fine, but `first_in_series` is flawed - it is written assuming that the release hasn't been cut yet, and the merge-back step happens post-release. Since the list of releases is refreshed every time, `first_in_series` will return False as soon as the first release has been cut, preventing this block from being entered.